channels/layers.py

```diff
@@ -155,19 +159,13 @@ def valid_channel_name(self, name, receive=False):
                         "Specific channel names in receive() must end at the !"
                     )
                 return True
-        raise TypeError(
-            "Channel name must be a valid unicode string containing only ASCII "
-            + "alphanumerics, hyphens, or periods, not '{}'.".format(name)
-        )
+        raise TypeError(self.invalid_name_error.format("Channel", name))
```

Comment:
Another approach is to define a custom exception, inheriting from a based, package-level exception class. This makes it easier to expect precisely in other calling code. But perhaps that would be for another day :p

Reply:
I like it a lot!